components/eamxx/src/share/field/field_manager.cpp

```diff
@@ -283,69 +332,50 @@ void FieldManager::registration_ends ()
   //     field that accommodate all bundled requests, it will return an empty list.
   //     Otherwise it will return the ordering of all fields in the cluster that allows all
   //     groups of the cluster to be a contiguous subset of C.
-  //  6) If step 4 fails for a cluster, remove from the cluster the groups whose bundling
-  //     is only "Preferred", and re-try.
-  //     Notes:
-  //      - eliminating 1+ groups from the cluster may actually "disconnect" the cluster,
-  //        into 2+ separate clusters, which can be treated separately. For simplicity,
-  //        we won't re build the cluster. It might make calling the function
-  //        contiguous_superset a bit more expensive, but it won't alter the existence
-  //        of a contiguous ordering.
-  //      - it is possible that, say, removing G1 still doesn't yield a "bundle-able"
-  //        cluster, but removing G2 does. In general, we should try to remove groups
-  //        one at a time, then two at a time, then three at a time,... until we
-  //        reach a point where the remaining groups can all be bundled. This is overly
-  //        complicated, so if 4 fails, we simply start removing groups with "Preferred"
-  //        bundling until 4 succeeds or we run out of groups with bundling=Preferred
-  //        to remove.
-  //
-
-  // Start by processing group request. This function will ensure that, if there's a
-  // request for group A that depends on the content of group B, the FieldGroupInfo
-  // for group A is updated to contain the correct fields, based on the content
-  // of group B. It also checks that the requests are not inconsistent.
-  pre_process_group_requests ();
+  //  Note: this can be done independent of grid since each grid will contain that same fields
+  //        in a group.
+
+  // Start by processing group request. This function checks that all group fields are properly
+  // registered on the appropriate grid and the FieldGroupInfo is up to date. If group G is
+  // requested, we make sure to register the union of all fields in group G on any grids they appear.
+  pre_process_group_requests();
 
   // Gather a list of groups to be bundled
-  // NOTE: copied groups are always created bundled, but in a second phase,
-  //       without creating individual subfields.
-  std::list<std::string> groups_to_bundle, copied_groups;
-  for (const auto& greqs : m_group_requests) {
-    for (const auto& r : greqs.second) {
-      if (r.derived_type!=DerivationType::Copy && r.bundling!=Bundling::NotNeeded) {
-        // There's at least one request for this group to be bunlded.
-        groups_to_bundle.push_back(greqs.first);
-      } else if (r.derived_type==DerivationType::Copy) {
-        copied_groups.push_back(greqs.first);
+  std::list<std::string> groups_to_bundle;
+  for (auto& grid_it : m_grids_mgr->get_repo()) {
+    for (const auto& greqs : m_group_requests.at(grid_it.second->name())) {
```

Comment:
I think here you could just have a loop over `m_group_request`, without the outer loop on the GM repo, no? Like
```c++
for (const auto& [grid_name, greqs] : m_group_requests) {
...
```

Reply:
I think in a follow up I'm going to remove storing `m_group_requests` since I believe I only need them for packsize info (everything else in the `m_field_groups` info. This would then go away so I decided to just leave it.